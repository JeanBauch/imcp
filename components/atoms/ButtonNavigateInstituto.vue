<template>
	<div :class="[themeColors.borderColor, themeColors.backgroundOnHover]" class="flex flex-col items-center justify-center px-9 py-8 md:px-12 md:py-11 lg:px-9 lg:py-8 xl:px-12 xl:py-11 rounded-[1.875rem] border bg-white cursor-pointer group">
		<div class="mb-4 md:mb-8 relative">
			<div v-if="icon.toLowerCase()" class="w-24 h-20 relative">
				<img :src="`icons/normal-colors/instituto/${icon}.svg`" class="block relative z-10" :alt="`icone ${title}`">
				<div class="absolute -inset-8 z-0 scale-[40%] md:scale-[60%]">
					<svg xmlns="http://www.w3.org/2000/svg" width="186" height="150" viewBox="0 0 186 150">
						<path
							id="palheta"
							:class="themeColors.backgroundPaletSVG"
							class="group-hover:fill-transparent transition-all ease-in-out"
							d="M73.724-8.847C30.131-8.847-36.2,4.243-40.255,52.624S48.427,141.153,92.02,141.153s53.6-30.144,53.6-73.616S117.317-8.847,73.724-8.847Z"
							transform="translate(40.384 8.847)"
						/>
					</svg>
				</div>
			</div>
		</div>
		<h3 :class="themeColors.textColor" class="capitalize font-ligth text-base md:text-lg 2xl:text-xl relative z-10 text-center">
			{{ title }}
		</h3>
	</div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
	data () {
		return {
			themeColors: {
				borderColor: `border-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-500`,
				backgroundOnHover: `hover:bg-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-50`,
				textColor: `text-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-500`,
				backgroundPaletSVG: `fill-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-50`,
			},
		};
	},
	props: {
		title: { type: String, required: true },
		icon: { type: String, required: true },
		color: {
			required: true,
			type: String,
			default: 'blue',
			validator: (value: string) => [
				'orange',
				'blue',
			].includes(value.toLowerCase()),
		},
	},
});
</script>
