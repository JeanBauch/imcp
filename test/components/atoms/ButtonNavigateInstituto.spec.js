import { shallowMount } from '@vue/test-utils';

import ButtonNavigateInstituto from '~/components/atoms/ButtonNavigateInstituto.vue';

describe('Components:atoms:ButtonNavigateInstituto', () => {
	const factory = (values = {}) => {
		return shallowMount(ButtonNavigateInstituto, {
			propsData: {
				...values,
			},
		});
	};
	describe('Snapshots', () => {
		it('should render component with correctly props', () => {
			const wrapper = factory({ title: 'Nossa história', icon: 'nossa-historia', color: 'orange' });

			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render correctly data as with props datas', () => {
			const wrapper = factory({ title: 'Nossa história', icon: 'nossa-historia', color: 'orange' });

			expect(wrapper.find('.border-primary-orange-500').exists()).toBeTruthy();
		});

		it('should render correctly SET data with color:orange', async () => {
			const wrapper = factory({ title: 'Nossa história', icon: 'nossa-historia', color: 'orange' });

			await wrapper.setData({ themeColors: { borderColor: 'border-primary-orange-500' } });
			await wrapper.setData({ themeColors: { backgroundOnHover: 'hover:bg-primary-orange-50' } });
			await wrapper.setData({ themeColors: { textColor: 'text-primary-orange-500' } });
			await wrapper.setData({ themeColors: { backgroundPaletSVG: 'fill-primary-orange-50' } });

			expect(wrapper.vm.themeColors.borderColor).toBe('border-primary-orange-500');
			expect(wrapper.vm.themeColors.backgroundOnHover).toBe('hover:bg-primary-orange-50');
			expect(wrapper.vm.themeColors.textColor).toBe('text-primary-orange-500');
			expect(wrapper.vm.themeColors.backgroundPaletSVG).toBe('fill-primary-orange-50');
		});

		it('should render correctly SET data with color:blue', async () => {
			const wrapper = factory({ title: 'O instituto', icon: 'instituto-inicial', color: 'blue' });

			await wrapper.setData({ themeColors: { borderColor: 'border-secondary-blue-500' } });
			await wrapper.setData({ themeColors: { backgroundOnHover: 'hover:bg-secondary-blue-50' } });
			await wrapper.setData({ themeColors: { textColor: 'text-secondary-blue-500' } });
			await wrapper.setData({ themeColors: { backgroundPaletSVG: 'fill-secondary-blue-50' } });

			expect(wrapper.vm.themeColors.borderColor).toBe('border-secondary-blue-500');
			expect(wrapper.vm.themeColors.backgroundOnHover).toBe('hover:bg-secondary-blue-50');
			expect(wrapper.vm.themeColors.textColor).toBe('text-secondary-blue-500');
			expect(wrapper.vm.themeColors.backgroundPaletSVG).toBe('fill-secondary-blue-50');
		});
	});
});
