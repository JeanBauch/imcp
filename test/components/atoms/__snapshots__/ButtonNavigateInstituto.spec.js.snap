// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:atoms:ButtonNavigateInstituto Snapshots should render component with correctly props 1`] = `
"<div class=\\"flex flex-col items-center justify-center px-9 py-8 md:px-12 md:py-11 lg:px-9 lg:py-8 xl:px-12 xl:py-11 rounded-[1.875rem] border bg-white cursor-pointer group border-primary-orange-500 hover:bg-primary-orange-50\\">
  <div class=\\"mb-4 md:mb-8 relative\\">
    <div class=\\"w-24 h-20 relative\\"><img src=\\"icons/normal-colors/instituto/nossa-historia.svg\\" alt=\\"icone Nossa história\\" class=\\"block relative z-10\\">
      <div class=\\"absolute -inset-8 z-0 scale-[40%] md:scale-[60%]\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"186\\" height=\\"150\\" viewBox=\\"0 0 186 150\\">
          <path id=\\"palheta\\" d=\\"M73.724-8.847C30.131-8.847-36.2,4.243-40.255,52.624S48.427,141.153,92.02,141.153s53.6-30.144,53.6-73.616S117.317-8.847,73.724-8.847Z\\" transform=\\"translate(40.384 8.847)\\" class=\\"group-hover:fill-transparent transition-all ease-in-out fill-primary-orange-50\\"></path>
        </svg></div>
    </div>
  </div>
  <h3 class=\\"capitalize font-ligth text-base md:text-lg 2xl:text-xl relative z-10 text-center text-primary-orange-500\\">
    Nossa história
  </h3>
</div>"
`;
